pkg/providers/cloudstack/validator.go

```diff
@@ -244,6 +245,23 @@ func (v *Validator) validateMachineConfig(ctx context.Context, datacenterConfig
 		if err = v.cmk.ValidateServiceOfferingPresent(ctx, zone.Id, machineConfig.Spec.ComputeOffering); err != nil {
 			return fmt.Errorf("validating service offering: %v", err)
 		}
+		if len(machineConfig.Spec.DiskOffering.Id) > 0 || len(machineConfig.Spec.DiskOffering.Name) > 0 {
```

Comment:
this validation seems redundant since it's already done in API context unware validation.

Reply:
refactored  diskoffering api validation so that it can be used else.